### PR TITLE
Re-implement next_after/prev_from logic to support jiff 0.2/DST

### LINKED ISSUE #26: Incorporate upstream DST fix
It looks like there's [a daylight saving time fix](https://github.com/zslayton/cron/pull/115) that's landed upstream. I ported the tests and they are failing (diff below). That said, I haven't looked closely enough at the fix to form an opinion re whether we should follow the same approach. More specifically, it would be nice if we could lean on `jiff` as much as possible to ensure DST correctness.

```diff
diff --git a/src/schedule.rs b/src/schedule.rs
index fe3f536..39dddeb 100644
--- a/src/schedule.rs
+++ b/src/schedule.rs
@@ -917,4 +917,53 @@ mod test {
         // But schedules yielding different events are not equal.
         assert_ne!(schedule_4, schedule_6);
     }
+
+    #[test]
+    fn test_dst_ambiguous_time_after() {
+        let schedule_tz = TimeZone::get("America/Chicago").unwrap();
+        let dt = DateTime::new(2022, 11, 5, 23, 30, 0, 0)
+            .unwrap()
+            .to_zoned(schedule_tz)
+            .unwrap();
+        let schedule = Schedule::from_str("0 0 * * * * *").unwrap();
+        let times = schedule
+            .after(&dt)
+            .map(|x| x.to_string())
+            .take(5)
+            .collect::<Vec<_>>();
+        let expected_times = [
+            "2022-11-06T00:00:00-05:00[America/Chicago]".to_string(),
+            "2022-11-06T01:00:00-05:00[America/Chicago]".to_string(),
+            "2022-11-06T01:00:00-06:00[America/Chicago]".to_string(), // 1 AM happens again
+            "2022-11-06T02:00:00-06:00[America/Chicago]".to_string(),
+            "2022-11-06T03:00:00-06:00[America/Chicago]".to_string(),
+        ];
+
+        assert_eq!(times.as_slice(), expected_times.as_slice());
+    }
+
+    #[test]
+    fn test_dst_ambiguous_time_before() {
+        let schedule_tz = TimeZone::get("America/Chicago").unwrap();
+        let dt = DateTime::new(2022, 11, 6, 3, 30, 0, 0)
+            .unwrap()
+            .to_zoned(schedule_tz)
+            .unwrap();
+        let schedule = Schedule::from_str("0 0 * * * * *").unwrap();
+        let times = schedule
+            .after(&dt)
+            .map(|x| x.to_string())
+            .rev()
+            .take(5)
+            .collect::<Vec<_>>();
+        let expected_times = [
+            "2022-11-06T03:00:00-06:00[America/Chicago]".to_string(),
+            "2022-11-06T02:00:00-06:00[America/Chicago]".to_string(),
+            "2022-11-06T01:00:00-06:00[America/Chicago]".to_string(),
+            "2022-11-06T01:00:00-05:00[America/Chicago]".to_string(), // 1 AM happens again
+            "2022-11-06T00:00:00-05:00[America/Chicago]".to_string(),
+        ];
+
+        assert_eq!(times.as_slice(), expected_times.as_slice());
+    }
 }
```

### PR DESCRIPTION
**Pull request by StephanvanSchaik**
Unfortunately, this is a rather big PR because I rewrote the logic in `Schedule::next_after()` and `Schedule::prev_from()` entirely to support the DST test cases from #26, while simultaneously ensuring that there are no regressions.

First, this PR introduces a number of helper functions to easily obtain the ordinal set given `Unit` (from the diff crate) as well as the current value for a corresponding unit given a `Zoned` object and `Unit`. `Schedule::adjust_next()` and `Schedule::adjust_prev()` are responsible for trying to pick the next ordinal for a given `Zoned` object and `Unit` and return either an updated `Zoned` object, or `None` if wraparound is required. Finally, `Schedule::reset_next()` and `Schedule::reset_prev()` reset the ordinal for a given `Zoned` object and `Unit` to the first/last ordinal.

`Schedule::next_after()` and `Schedule::prev_from()` now build upon these convenience functions. More specifically, they first determine the smallest possible unit for which we can simply pick the next ordinal, while ensuring that the larger units already contain valid ordinals. Otherwise we have to adjust those as well. Once we have found the smallest possible unit, we reset all smaller units to the first/last ordinals.

There are a number of edge cases which are obvious from running `cargo test`. First, the largest possible ordinal for the day unit specifically can be larger than the number of days in the month. So we must ensure that we pick an ordinal that is actually valid. Furthermore, it is possible that the timestamp contains a non-zero unit smaller than seconds (e.g. milliseconds, microseconds, nanoseconds). To prevent this from causing issues, we first try rounding up or down to the nearest second and see if the rounded timestamp adheres to the given `Schedule`. In the case that it is a valid timestamp, we simply return the rounded timestamp. This pretty much ensures that all tests pass.

In addition to ensuring that `cargo test` passes, I have also fixed all `cargo clippy` lints. In particular, the upside of this PR is that we can now do without `src/queries.rs` entirely and that the overall logic is slightly simpler and as result we actually end up with less code in jiff-cron :).

However, reviews are definitely welcome (@maxcountryman @LeoniePhiline) since I went through multiple attempts today before I landed on this solution. Landing this PR should close #26 and #29, and prepare #29 for merging. If the indirection is annoying, let me know, I can have a look at opening a PR against the main branch (I just continued working on the renovate branch for jiff 0.2).

Finally, I also looked at the rest of jiff's changelog, but it seems like the DST issue is the most obvious issue that needs addressing. I didn't see anything else of concern, but it would be good to have a quick look to make sure I didn't miss anything there.

### CODE REVIEW

**Review by maxcountryman — APPROVED** [review]

**Comment by maxcountryman** [comment]
Always love to see more code deleted than added. 🙂

**Comment by maxcountryman** [comment]
I reviewed the jiff changes as well and nothing jumps out as concerning to me. I think we can go ahead and merge this.

**Comment by LeoniePhiline** [comment]
Thanks so much @StephanvanSchaik!
